kratos/python/add_variable_utils_to_python.cpp

```diff
@@ -452,6 +452,63 @@ void AddVariableUtilsToPython(pybind11::module &m)
         .def("ApplyFixity", ApplyFixity<Variable<double>>)
         .def("ApplyFixity", ApplyFlaggedFixity<Variable<double>>)
         .def("ApplyVector", &VariableUtils::ApplyVector<Variable<double>>)
+        .def("ApplyVector", []( VariableUtils& rVariableUtils, 
+                                const Variable<array_1d<double,3>>& rVar,
+                                unsigned int Dim,
+                                const Vector& rData,
+                                ModelPart::NodesContainerType& rNodes
+                                )
+                {
+                    if(rNodes.size() != 0 && rNodes.size()*Dim == rData.size()) {
+                        // First we do a check
+                        rVariableUtils.CheckVariableExists(rVar, rNodes);
+
+                        IndexPartition<std::size_t>(rNodes.size()).for_each([&](std::size_t index){
+                            auto it_node = rNodes.begin() + index;
+                            auto& v = it_node->FastGetSolutionStepValue(rVar);
+                            for(unsigned int k=0; k<Dim; k++)
```

Comment:
Why do you need dim, when you can deduce the dime from the vector?

Reply:
Ah, Ok, because the data contains the values for all the nodes.... But in that case you can deduce the size from the fact that you are not considering Variable<Vector>, but  array_1d<double, 3>, and therefor the size must be 3. Except if you want to customize for 2D vectors...

Reply:
I'd extract the Dimension from `rData.size() / rNodes.size()`. Having `Dim` as an argument seems superfluous.

Reply:
no, i am adding the dim on purpose. the point is that when i am in 2D i want to obtain only the xy component. this is very useful for example to write a rotation in 2D